rust/core-lib/src/editor.rs

```diff
@@ -879,14 +879,15 @@ impl Editor {
         }
         let mut spans = sb.build();
         if rev != self.engine.get_head_rev_id().token() {
-            let delta = self.engine.delta_rev_head(rev);
-            let mut transformer = Transformer::new(&delta);
-            let new_start = transformer.transform(start, false);
-            if !transformer.interval_untouched(Interval::new(start, end_offset)) {
-                spans = spans.transform(start, end_offset, &mut transformer);
+            if let Ok(delta) = self.engine.try_delta_rev_head(rev) {
+                let mut transformer = Transformer::new(&delta);
+                let new_start = transformer.transform(start, false);
+                if !transformer.interval_untouched(Interval::new(start, end_offset)) {
+                    spans = spans.transform(start, end_offset, &mut transformer);
+                }
+                start = new_start;
+                end_offset = transformer.transform(end_offset, true);
```

Comment:
I think it's worth logging the error, since it's still definitely unexpected behaviour and probably indicates programmer or user error _somewhere_.